File: PatternFrontEnd/patterns-manager/src/app/patterns/_models/pattern.ts
import { Base } from "src/app/shared/_models/base";
import { PatternCompany } from 'src/app/settings/_models/pattern-company';
import { PatternType } from './../../settings/_models/pattern-type';

export interface Pattern extends Base {
  name?:string;
  patternTypeId?: number;
  patternType?: PatternType;
  patternCompanyId?: number;
  patternCompany?: PatternCompany;
}

File: PatternAPI/PatternAPI/Models/Pattern.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace PatternsAPI.Models
{
    public class Pattern : BaseEntity
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public string? Name { get; set; }
        public int PatternTypeId { get; set; }
        public PatternType? PatternType { get; set; }
        public int PatternCompanyId { get; set; }
        public PatternCompany? PatternCompany { get; set; }
        public string? ImgUrl { get; set; }

        public string GetPatternTypeName()
        {
            return PatternType?.Name ?? "";
        }

        public string GetPatternCompanyName()
        {
            return PatternCompany?.Name ?? "";
        }
    }
}